### DISCUSSION TITLE
GPU acceleration for `PxScene::overlap()`

### DISCUSSION THREAD

**Discussion by SumantBagri** [NVIDIA-Omniverse/PhysX#179] [Q&A] [answered]
Hi,

Does the `PxScene::overlap()` [query](https://nvidia-omniverse.github.io/PhysX/physx/5.2.1/docs/SceneQueries.html#overlaps) benefit from gpu acceleration ? Or is the gpu acceleration currently only provided for `PxScene::simulate()` ?

**Comment by preist-nvidia** [comment, marked as answer]
Hi Sumant,

No, scene queries are currently CPU only.

Philipp

**Reply by SumantBagri** [reply]
Hi Philipp,

Thank you for your response!

Is GPU support for scene queries planned for the future ?

**Reply by preist-nvidia** [reply]
We don't have a timeline for GPU scene queries. Are you hitting perf bottlenecks with CPU queries or why are you interested in GPU queries?

**Reply by SumantBagri** [reply]
I wanted to leverage the GPU acceleration provided PhysX for collision checking during robot motion planning and benchmark its performance. I am able to execute collision checking on the GPU using `simulate()` but it seemed like it would be easier to use `scene queries` instead for this specific task as I am not concerned about the physics of the collision. That's why I wanted to know if GPU queries are supported. It is purely from a research point-of-view.

Anyways, thanks for letting me know Philipp.

P.S - Would you happen to know if it is possible to generate contacts for collisions between two triangle mesh geometries using the `ContactReportCallback` ? (https://github.com/NVIDIA-Omniverse/PhysX/discussions/180)

**Reply by preist-nvidia** [reply]
@SumantBagri - I agree that scenequery is the right tool here. But please give the CPU scenequery a shot, it is fast.

For contact gen, you can try the immediate mode: https://nvidia-omniverse.github.io/PhysX/physx/5.2.1/_build/physx/latest/function_group__immediatemode_1ga005fa8fe54245ea3fec26d17f14874a9.html#exhale-function-group-immediatemode-1ga005fa8fe54245ea3fec26d17f14874a9

**Reply by SumantBagri** [reply]
Thanks for your suggestions @preist-nvidia. Much appreciated :). I will try them out
